From d1fe3b29bf7ff8204624761885578cf7e162b903 Mon Sep 17 00:00:00 2001
From: Jin Xue <jin.xue@intel.com>
Date: Fri, 30 Aug 2019 09:57:25 +0800
Subject: [PATCH 5/6] use the default subdevice

use the default subdevice in the primary HAL

Change-Id: I85dc805fac9db17606f4de9e7e2b49ec8cb7e9ec
Tracked-On: ACP-740
Signed-off-by: Jin Xue <jin.xue@intel.com>
Reviewed-on: 679177
Reviewed-by: Yang, Bin Y <bin.y.yang@intel.com>
---
 Android.mk |  2 ++
 audio_hw.c | 13 +------------
 2 files changed, 3 insertions(+), 12 deletions(-)

diff --git a/Android.mk b/Android.mk
index ccf2807..80b7349 100644
--- a/Android.mk
+++ b/Android.mk
@@ -35,6 +35,8 @@ LOCAL_C_INCLUDES += \
 
 LOCAL_MODULE := audio.primary.$(TARGET_PRODUCT)
 
+LOCAL_PROPRIETARY_MODULE := true
+
 LOCAL_MODULE_RELATIVE_PATH := hw
 
 LOCAL_MODULE_TAGS := optional
diff --git a/audio_hw.c b/audio_hw.c
index 1178710..04846fa 100644
--- a/audio_hw.c
+++ b/audio_hw.c
@@ -52,11 +52,6 @@
 #define IN_PERIOD_COUNT 2
 #define IN_SAMPLING_RATE 48000
 
-#define CONNECTED_DISPLAYS 3
-#define DISPLAY_SHIFT 0
-
-unsigned int hdmi_output[CONNECTED_DISPLAYS] = {3, 7, 8};
-
 struct pcm_config pcm_config_out = {
     .channels = 2,
     .rate = OUT_SAMPLING_RATE,
@@ -192,13 +187,7 @@ static int start_output_stream(struct stream_out *out)
         return -ENODEV;
     }
 
-    char prop_id[PROPERTY_VALUE_MAX];
-    memset(prop_id, 0, sizeof(prop_id));
-    property_get("sys.container.id", prop_id, "0");
-    unsigned int id = atoi(prop_id) % CONNECTED_DISPLAYS;
-    unsigned int device = hdmi_output[(id + DISPLAY_SHIFT) % CONNECTED_DISPLAYS];
-
-    out->pcm = pcm_open(PCM_CARD, device, PCM_OUT | PCM_NORESTART | PCM_MONOTONIC, out->pcm_config);
+    out->pcm = pcm_open(PCM_CARD, PCM_DEVICE, PCM_OUT | PCM_NORESTART | PCM_MONOTONIC, out->pcm_config);
     if (!out->pcm) {
 	ALOGE("pcm_open(out) failed: device not found");
         return -ENODEV;
-- 
2.7.4

